V2.16-RUN-15152-alertmanager-alerts (#633)
run-ai/docs: docs/admin/troubleshooting/alertmanager/RunaiDeploymentUnavailableReplicas.md

---
title: RunaiDeploymentUnavailableReplicas 
summary: This article describes the RunaiDeploymentUnavailableReplicas alert.
authors:
    - Jason Novich
    - Viktor Koukouliev
date: 2024-Jan-10
---

## Meaning

> The Run:ai deployment has one or more unavailable pods.

Reviewer: ```suggestion
A Run:ai deployment has one or more unavailable pods.
```

## Impact

> This indicates scaling issues. The new version deployment can not deploy due to missing features.

Reviewer: ```suggestion
This alert may indicate scaling issues.
If a new version of the deployment can not be deployed, this may cause missing new features.
```

Author: ```suggestion
This alert may indicate scaling issues.
If a new version of the deployment can not be deployed, this may cause new features to be missing.
```

## Severity

Warning

## Diagnosis

> Run `kubectl get deployment -n runai kubectl get deployment -n runai-backend` to see if one or more of the deployments is missing pods.

Reviewer: ```suggestion
Run 
`kubectl get deployment -n runai`
` kubectl get deployment -n runai-backend` 

Identify one or more deployments that have missing pods.
```

## Mitigation

> Run `kubectl describe deployment X -n runai/runai-backend kubectl describe replicaset X -n runai/runai-backend kubectl logs deployment/X -n runai/runai-backend` to try and figure out why the deployment cannot create pods. Contact Run:ai if you cannot correct the issue.

Reviewer: ```suggestion
Run 
`kubectl describe deployment X -n runai/runai-backend 
kubectl describe replicaset X -n runai/runai-backend 
kubectl logs deployment/X -n runai/runai-backend` 

From the logs and deployment details, try and figure out why the deployment cannot create pods. 
If you cannot correct the issue, contact Run:ai support.
```
